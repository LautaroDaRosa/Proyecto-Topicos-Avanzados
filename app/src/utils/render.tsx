import React from 'react';
import { render } from '@testing-library/react';

import { MemoryRouter } from 'react-router-dom';
import { HelmetProvider } from 'react-helmet-async';
import { Provider } from 'react-redux';
import { configureAppStore } from 'store/configureStore';

const store = configureAppStore();

const renderWithWrappers = (content: React.ReactNode) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>
        <HelmetProvider>{content}</HelmetProvider>
      </MemoryRouter>
      ,
    </Provider>,
  );
};

export default renderWithWrappers;
