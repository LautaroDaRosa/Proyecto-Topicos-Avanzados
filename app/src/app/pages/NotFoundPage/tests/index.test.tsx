import React from 'react';
import { NotFoundPage } from '..';
import render from 'utils/render';
import { screen } from '@testing-library/react';

describe('NotFoundPage', () => {
  it('should render correctly', () => {
    const { container } = render(<NotFoundPage />);
    expect(container).toMatchSnapshot();

    screen.getByText('Page not found.');
    screen.getByRole('link', { name: 'Return to Home Page' });
  });
});
