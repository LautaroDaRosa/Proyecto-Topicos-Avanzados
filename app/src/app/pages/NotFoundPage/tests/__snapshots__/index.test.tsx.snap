// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should render correctly 1`] = `
.c2 {
  font-size: 1rem;
  line-height: 1.5;
  margin: 0.625rem 0 1.5rem 0;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c3:active {
  opacity: 0.4;
}

.c0 {
  height: calc(100vh - 4rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 320px;
}

.c1 {
  margin-top: -8vh;
  font-weight: bold;
  font-size: 3.375rem;
}

.c1 span {
  font-size: 3.125rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      4
      <span
        aria-label="Crying Face"
        role="img"
      >
        ðŸ˜¢
      </span>
      4
    </div>
    <p
      class="c2"
    >
      Page not found.
    </p>
    <a
      class="c3"
      href="/"
    >
      Return to Home Page
    </a>
  </div>
  ,
</div>
`;
